src/main/java/com/redislabs/redistimeseries/RedisTimeSeries.java

```diff
@@ -357,29 +310,33 @@ public long add(String sourceKey, long timestamp, double value, long retentionTi
    */
   public long add(String sourceKey, long timestamp, double value, long retentionTime, boolean uncompressed, Map<String, String> labels) {
     try (Jedis conn = getConnection()) {
+      byte[][] args = tsAddArgs(sourceKey, timestamp, value, retentionTime, uncompressed, labels);
+      return sendCommand(conn, Command.ADD, args).getIntegerReply();
+    }
+  }
 
-      byte[][] args = new byte[5 + (labels==null ? 0 : 2*labels.size()+1) + (uncompressed?1:0)][];
-      int i=0;
+  private static byte[][] tsAddArgs(String sourceKey, long timestamp, double value, long retentionTime, boolean uncompressed, Map<String, String> labels) {
+    byte[][] args = new byte[3 + (retentionTime>=0 ? 2 : 0 ) + (labels==null ? 0 : 2*labels.size()+1) + (uncompressed?1:0)][];
+    int i=0;
 
-      args[i++] = SafeEncoder.encode(sourceKey);
-      args[i++] = timestamp>0 ? Protocol.toByteArray(timestamp) : STAR;
-      args[i++] = Protocol.toByteArray(value);
+    args[i++] = SafeEncoder.encode(sourceKey);
+    args[i++] = timestamp>0 ? Protocol.toByteArray(timestamp) : STAR;
+    args[i++] = Protocol.toByteArray(value);
+    if(retentionTime>=0){
```

Comment:
this is a behavior change and will hide user's mistakes (giving negative numbers)